src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -555,6 +570,15 @@ def get(self, args):
         # Check the command index for (command: [module]) mapping, like
         # "network": ["azure.cli.command_modules.natgateway", "azure.cli.command_modules.network", "azext_firewall"]
         index_modules_extensions = index.get(top_command)
+        if not index_modules_extensions and is_autocomplete():
+            # If user type `az acco`, command begin with `acco` will be matched.
+            # If the user just installed a new extension which contains a command named `acco`, he can't get the
+            # expected suggestion. But it's a rare scenario, and it's okay to ignore.
```

Comment:
This does not happen as `CommandIndex().invalidate()` is called after extension installation/remove/update.